sdl2promela/translator.py

```diff
@@ -544,66 +559,192 @@ def __get_exit_procedures(
+        return False
 
-        for procedure_name in exit_procedures:
-            statements.append(CallBuilder().withTarget(procedure_name).build())
+    input_block = input.transitions[key]
+    statements: List[promelamodel.Statement] = []
 
+    # Generate assignment of signal parameters into variables
+    for target_variable_ref in input_block.target_variables:
+        variable = context.sdl_model.variables[target_variable_ref.variableName]
+        variableType = resolve_asn1_type(context.sdl_model.types, variable[0])
+        assignInlineName = __get_assign_value_inline_name(variableType)
         statements.append(
             CallBuilder()
-            .withTarget(transition_function_name)
+            .withTarget(assignInlineName)
+            .withParameter(__generate_variable_name(context, target_variable_ref, True))
             .withParameter(
-                VariableReferenceBuilder(str(input_block.transition_id)).build()
+                VariableReferenceBuilder(
+                    __get_parameter_name(input.parameters[0].target_variable)
+                ).build()
             )
             .build()
         )
-        switch_builder.withAlternative(
-            AlternativeBuilder()
-            .withCondition(
-                BinaryExpressionBuilder(promelamodel.BinaryOperator.EQUAL)
-                .withLeft(VariableReferenceBuilder(state_variable_name).build())
-                .withRight(
-                    VariableReferenceBuilder(__get_state_name(context, state)).build()
+
+    # Generate exit procedure calls in case of composite state.
+    exit_procedures = __get_exit_procedures(context, input_block, key)
+
+    for procedure_name in exit_procedures:
+        statements.append(CallBuilder().withTarget(procedure_name).build())
+
+    # Generate an alternative for switch,
+    # which checks current state and executes transition
+    statements.append(
+        CallBuilder()
+        .withTarget(transition_function_name)
+        .withParameter(VariableReferenceBuilder(str(input_block.transition_id)).build())
+        .build()
+    )
+
+    builder.withAlternative(
+        AlternativeBuilder()
+        .withCondition(
+            BinaryExpressionBuilder(promelamodel.BinaryOperator.EQUAL)
+            .withLeft(VariableReferenceBuilder(state_variable_name).build())
+            .withRight(VariableReferenceBuilder(__get_state_name(context, key)).build())
+            .build()
+        )
+        .withStatements(statements)
+        .build()
+    )
+
+    return True
+
+
+def generate_transition_state_switch(
+    context: Context,
+    statename: str,
+    input: sdlmodel.Input,
+    builder: SwitchBuilder,
+    parentstate: Optional[str],
+) -> bool:
+    """
+    Recursive function, responsible for generation of alternative for switch,
+    which should finally call 'execute transition' with proper transition id.
+    In case of parallel states, the parameter 'substate_name' shall be not None,
```

Comment:
Is there such a parameter - substate_name?

Reply:
changed